### PR TITLE
doc: Document or eliminate panics

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Potential panics should be documented.

# Changes
- Stop ignoring linter errors for missing panic documentation.
- Where panics can be eliminated in a couple of lines of code, do so.
- Where panics are in fact not possible or require more memory than the state of the universe, use `unreachable!("some explanation")` to communicate this.
- Document the remaining panics.

# Tests
Existing CI should suffice

# Todos

- [ ] Add entry to changelog (if necessary).

### CODE REVIEW

File: rs/backend/src/accounts_store.rs [hunk 1]
```diff
@@ -1013,29 +1028,34 @@ impl AccountsStore {
                 })
                 .collect();
```
Comment (bitdivine) [inline rs/backend/src/accounts_store.rs:1030]: Below I changed the `unwrap()` into `if let Some(..) = `

File: rs/backend/src/accounts_store.rs [hunk 2]
```diff
@@ -1649,14 +1669,26 @@ impl Account {
...
             .transactions
             .push(transaction_index);
     }
 
+    /// Records a transaction in a hardware wallet account.
+    ///
+    /// TODO: Use the index canister instead.
+    ///
+    /// # Panics
+    /// - If the account does not have a hardware wallet sub-account with the given identifier.
```
Comment (przydatek) [inline rs/backend/src/accounts_store.rs:1691]: nit: maybe change the `unwrap()` in this function to an `expect()`?

File: rs/backend/src/assets.rs [hunk 3]
```diff
@@ -386,6 +386,9 @@ pub fn init_assets() {
 /// Note: The `Vec` is mutated during decompression, so pass by reference is inefficient
 ///       as it would force the data to be copied into a new vector, even when the
 ///       original is no longer needed.
+///
+/// # Panics
+/// - If the decompression fails or the tarball cannot be parsed.
```
Comment (przydatek) [inline rs/backend/src/assets.rs:391]: nit: same here, `unwrap` -> `expect`?

**Review by przydatek — APPROVED** [review]
LGTM, thanks!

File: rs/backend/src/assets.rs (hunk 3 above)
Comment (bitdivine) [reply rs/backend/src/assets.rs:391]: Good idea.  Done.  Even better would be `ic_cdk::trap(format!("Explanation with filename"))` but  I think this is enough for this PR.

**Pull request merged**